content/articles/spring-boot-recaptcha-v3/index.md

```diff
@@ -0,0 +1,855 @@
+### Implementing Google reCAPTCHA in a SpringBoot application
+
+### Introduction
+reCAPTCHA is a system that enables users to protects their sites from bots. ReCaptcha generates a score that ranges from `0` to `1`. If the score is less than `0.5`, there is a possibility a bot invoked the action, and if the score is greater than `0.5`, the action was not invoked by a bot. This tutorial will teach you how to Implement Google reCAPTCHA version 3 to protect a login form using SpringBoot. We will intercept the authentication during login. If the score generated is less than `0.5`, the user will be requested to enter an OTP sent to their email, or the authentication will continue processing. 
+
+
+### Table of Contents
+- [Create new site with Google to verify your captcha response]()
+- [Create a SpringBoot application using spring initializr]()
+- [Configure database connection properties]()
+- [Create User, Role, and ReCaptchaResponse models]()
+- [Create User and Role repositories]()
+- [Add a `CommandLineRunner` to create custom users]()
+- [Create a user service to handle user details]()
+- [Add `WebSecurityConfigurerAdapter` configuration class to handle login ]()
+- [Create ReCAPTCHA handler to get the score]()
+- [Create a Login Filter to redirect the user based on the score]()
+- [Add the `addFilterBefore()` to `HttpSecurity`]()
+- [Create a login and OTP Controller]()
+- [Create a login and OTP form and a default success page]()
+- [Test the login when the score is greater than 0.5]()
+- [Test the login when the score is less than 0.5]()
+
+### Prerequisites
+- Spring Boot
+- Spring Data JPA
+- Spring Security
+- Thymeleaf
+- JDK 11+
+- Intellij IDE
+
+### Create a new site with Google to verify your captcha response
+[Create a new reCAPTCHA V3 site ](https://www.google.com/recaptcha/admin/create) with any name you want, but the domain can be localhost as we will be testing the application locally. Take note of the site key and secret key, which we will use later in the application.
+
+### Create a SpringBoot application using spring initializr
+Go to [Spring Initialzr](https://start.spring.io/) and generate a new project with dependency `Spring Web`, `Spring Security`, `Spring Data JPA`, `Thymeleaf`, `Spring Boot Dev Tools`, `MySQL Driver` and `Lombok`.
+
+Import the generated project into Intellij and ensure you have an active internet connection to download the dependencies from the remote central repository.
+
+![Spring Initializr image](spring-initializr.png)
+
+### Configure database connection properties
+Since we will not deploy the application to production, the database tables, and data will be automatically created for us during application initialization using `spring.jpa.hibernate.ddl-auto=create` property.
```

Comment:
change to "Since we will not deploy the application to production, the database tables, and data will be automatically created during application initialization using `spring.jpa.hibernate.ddl-auto=create` property."